Firestore/core/test/firebase/firestore/model/field_value_test.cc

```diff
@@ -144,6 +144,20 @@ TEST(FieldValue, BlobType) {
   EXPECT_FALSE(a < a);
 }
 
+TEST(FieldValue, ReferenceType) {
+  const DatabaseId id("project", "database");
+  const FieldValue a =
+      FieldValue::ReferenceValue(DocumentKey::FromPathString("abc"), &id);
+  DocumentKey key = DocumentKey::FromPathString("def");
+  const FieldValue b = FieldValue::ReferenceValue(key, &id);
+  const FieldValue c = FieldValue::ReferenceValue(std::move(key), &id);
```

Comment:
Optional: consider checking that key was actually moved (`EXPECT_TRUE(key.path().empty())` or similar).

Reply:
That's implementation details of DocumentKey, which I don't want to test in the field_value_test here.